File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int res = nums[0];
        int L = 0;
        int R = nums.size() - 1;

        while(L <= R){
            if(nums[L] < nums[R]){
                res = min(res, nums[L]);
            }

            int M = (L + R) / 2;
            res = min(res, nums[M]);

            if(nums[L] <= nums[M]){
                L = M + 1;
            }else{
                R = M - 1;
            }
        }    

        return res;
    }
};

File: Data Structures & Algorithms/sqrtx/submission-3.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x
        res = 0

        while l <= r:
            m = l + ((r -l) // 2)
            sqrd = m ** 2
            if sqrd == x:
                return m
            elif sqrd < x:
                l = m + 1
                res = m
            else:
                # sqrd > x:
                r = m - 1
        
        return res